easybuild/framework/easyblock.py

```diff
@@ -4928,106 +4930,142 @@ def get_updated_exts_list(exts_list, exts_defaultclass, bioconductor_version=Non
 
-            # close the extension
-            exts_list_formatted.append('%s}),' % (INDENT_4SPACES,))
+                    # append the key and value of the option
+                    exts_list_formatted.append("%s'%s': %s," % (INDENT_4SPACES * 2, key, value))
 
-    # close the exts_list
-    exts_list_formatted.append(']\n')
+                # close the extension
+                exts_list_formatted.append('%s}),' % (INDENT_4SPACES,))
 
-    # read the easyconfig file and replace the exts_list with the new one
-    regex = re.compile(r'^exts_list(.|\n)*?\n\]\s*$', re.M)
-    ectxt = regex.sub('\n'.join(exts_list_formatted), read_file(path))
+        # close the exts_list
+        exts_list_formatted.append(']\n')
 
-    # write the new easyconfig file
-    write_file(path, ectxt)
+        # read the easyconfig file and replace the exts_list with the new one
+        regex = re.compile(r'^exts_list(.|\n)*?\n\]\s*$', re.M)
+        new_ec = regex.sub('\n'.join(exts_list_formatted), read_file(ec['spec']))
+    else:
+        raise EasyBuildError("Invalid parameter to update Easyconfig")
+
+    return new_ec
 
 
 def get_exts_list(ec):
     """
-    Get the extension list from an EasyConfig instance.
+    Get the extension list from the given EasyConfig instance.
 
-    :param ec: EasyConfig instance
+    :param ec: EasyConfig instance.
 
-    :return: exts_list of the given EasyConfig instance
+    :return: list of extensions from the given EasyConfig instance.
     """
 
-    exts_list = ec.get('ec', {}).get('exts_list', None)
+    if not ec:
+        raise EasyBuildError("No EasyConfig instance provided to retrieve extensions from")
 
-    if not exts_list:
-        raise EasyBuildError("No extension list found in easyconfig")
+    # get the extension list from the easyconfig file
+    exts_list = ec.get('ec', {}).get('exts_list', [])
+
+    if exts_list:
+        print_msg("Found %s extensions..." % len(exts_list), log=_log)
+    else:
+        print_warning("No extensions found in easyconfig...", log=_log)
 
     return exts_list
 
 
 def get_exts_list_class(ec):
     """
-    Get the exts_defaultclass or deduct it from an EasyConfig instance.
+    Get the exts_defaultclass or deduce it from the given EasyConfig instance.
 
-    :param ec: EasyConfig instance
+    :param ec: EasyConfig instance.
 
-    :return: the exts_defaultclass of the given EasyConfig instance
+    :return: the class of the extensions from the given EasyConfig instance.
     """
 
+    if not ec:
+        raise EasyBuildError("No EasyConfig instance provided to retrieve extensions from")
+
+    # get the extension list class from the easyconfig file
     exts_list_class = ec.get('ec', {}).get('exts_defaultclass', None)
+
+    # if no exts_defaultclass is found, try to deduce it from the EasyConfig name
     if not exts_list_class:
+        
+        # get the name of the EasyConfig
         name = ec.get('ec', {}).get('name', None)
+
         if name:
```

Comment:
Use this option to define exts_defaultclass for PythonBundles: Taken from PythonBundle EasyBlock
If ec.easyblock=PythonBundle, 
elf.cfg['exts_defaultclass'] = 'PythonPackage'

Reply:
This is to use the same mechanism that PythonBundle EasyBlocks use to define exts_defaultclass value